Src/FluentAssertions/AsyncAssertionsExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System.Threading.Tasks;
+using FluentAssertions.Specialized;
+
+namespace FluentAssertions
+{
+    public static class AsyncAssertionsExtensions
+    {
+#pragma warning disable AV1755 // "Name of async method ... should end with Async"; Async suffix is too noisy in fluent API
+        /// <summary>
+        /// Asserts that the completed <see cref="TaskCompletionSource{TResult}"/> provides the specified result.
+        /// </summary>
+        /// <param name="task">The <see cref="TaskCompletionSourceAssertions{T}"/> containing the <see cref="TaskCompletionSource{TResult}"/>.</param>
+        /// <param name="expected">The expected value.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        /// <typeparam name="T"></typeparam>
+        public static async Task WithResult<T>(this Task<AndWhichConstraint<TaskCompletionSourceAssertions<T>, T>> task,
```

Comment:
Should this return a `Task<AndWhichConstraint<TaskCompletionSourceAssertions<T>, T>>`?

Reply:
This should be possible.
What is the example (test) you would like to build?

Reply:
My thought was to allow further chaining.
I don't have a concrete example at hand.

Reply:
While trying to implement that I found that with the option to chain these assertions I run into same problem I had before implementing this extension: I need to await the result before chaining.

I feel we need a new `AsyncAndContraint` or similar to chain the await of the `Task`s, isn't it?
For now I stopped to implement this option.

Reply:
Finally I changed my mind...
Even the explicit use of the result (with `(await ...).And...`) is not fluent api, but returning the constraint allows other extension methods, I think.